### PR TITLE
Import beauty/1.0.0-rc1

### PR DESCRIPTION
**Pull request by dfleury2**
beauty/1.0.0-rc1

This is a new project to easily create HTTP REST server in C++

I am the developper and maintainer of Beauty.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/beauty/all/conanfile.py
Comment (dvirtz) [inline recipes/beauty/all/conanfile.py:59]: if you don't customize the generators you can just have
```
generators = "CMakeDeps", "CMakeToolchain"
```
as a class property

File: recipes/beauty/all/conanfile.py
Comment (dvirtz) [inline recipes/beauty/all/conanfile.py:77]:
```suggestion
        cmake = CMake(self)
        cmake.install()
```

File: recipes/beauty/all/conanfile.py
Comment (dvirtz) [inline recipes/beauty/all/conanfile.py:80]:
```suggestion
```
this is the default

File: recipes/beauty/all/conanfile.py
Comment (dfleury2) [reply recipes/beauty/all/conanfile.py:77]: This is how I done it at first, but this trigger a error because it's installing some cmake specific files and the hook complains about them.

File: recipes/beauty/all/conanfile.py
Comment (dfleury2) [reply recipes/beauty/all/conanfile.py:80]: Ok, I will try when I remove this default. I do not have components this should be fine.

File: recipes/beauty/all/conanfile.py
Comment (dvirtz) [reply recipes/beauty/all/conanfile.py:77]: the idiom is to install and then delete the unwanted files
```
        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```
https://github.com/conan-io/conan-center-index/blob/master/docs/package/all/conanfile.py#L122

Comment (dfleury2) [reply recipes/beauty/all/conanfile.py]: Just push all the changes

**Review by SSE4 — DISMISSED** [review]

File: recipes/beauty/all/conanfile.py
Comment (dvirtz) [inline recipes/beauty/all/conanfile.py:72]: there's a [warning](https://github.com/conan-io/conan-center-index/pull/10956#issuecomment-1138904415) about a missing system dependency

```suggestion
        self.cpp_info.requires = ["boost::headers", "openssl::openssl"]
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs = ["m"]
```

File: recipes/beauty/all/conanfile.py
Comment (toge) [inline recipes/beauty/all/conanfile.py:39]:
```suggestion
            tools.check_min_cppstd(self, "20")
```

beauty seems to require C++20.
https://github.com/dfleury2/beauty/blob/master/CMakeLists.txt#L25

File: recipes/beauty/all/conanfile.py
Comment (toge) [inline recipes/beauty/all/conanfile.py:67]: For feature release, please call `configure()` before `install()`.
And if possible, please consider create `_configure_cmake()` like this.
https://github.com/conan-io/conan-center-index/blob/master/recipes/trantor/all/conanfile.py#L89-L107

Comment (dfleury2) [reply recipes/beauty/all/conanfile.py:67]: ok

File: recipes/beauty/all/conanfile.py
Comment (dfleury2) [reply recipes/beauty/all/conanfile.py]: Done.
I will fix the CMakeLists.txt on next release

File: recipes/beauty/all/conanfile.py
Comment (dfleury2) [reply recipes/beauty/all/conanfile.py]: Done

File: recipes/beauty/all/conanfile.py
Comment (dvirtz) [reply recipes/beauty/all/conanfile.py:67]: I thought it's not needed with the new CMake build helper

Comment (dfleury2) [reply recipes/beauty/all/conanfile.py:67]: I thought that too... but here I will wait for the approval before anything

Comment (toge) [reply recipes/beauty/all/conanfile.py:67]: I think it is not important too.
I will follow your decision.

**Review by toge — DISMISSED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/beauty/all/conanfile.py
Comment (uilianries) [reply recipes/beauty/all/conanfile.py:67]: This is the future! We no longer need that cmake wrapper + cmake generator. It will be deprecated in Conan 2.0. Note that this recipe is using the CMake helper from `conan.tools.cmake`, and not from `conans`. They are incompatible. More https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmake.html.
This approach is correct and should be adopted with the Conan 2.0 transition.
Also, it's documented [here](https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-generate-method). The CCI has v2 documented, but only related to cpp_info.

File: recipes/beauty/all/conanfile.py
Comment (uilianries) [inline recipes/beauty/all/conanfile.py:40]:
```suggestion
    def config_options_self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def configure(self):
```
self.settings values are available during config_options. However, self.options, are only evaluated after that.

Comment (dfleury2) [reply recipes/beauty/all/conanfile.py]: Done

File: recipes/beauty/all/conanfile.py
Comment (uilianries) [inline recipes/beauty/all/conanfile.py:60]:
```suggestion
        if is_msvc and self.options.shared:
```

Because it can be both 'Visual Studio' or 'msvc'

File: recipes/beauty/all/conanfile.py
Comment (uilianries) [inline recipes/beauty/all/conanfile.py:3]:
```suggestion
from conan.tools.cmake import CMake
from conan.tools.microsoft import is_msvc
```
https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-is-msvc

File: recipes/beauty/all/test_package/CMakeLists.txt
Comment (uilianries) [inline recipes/beauty/all/test_package/CMakeLists.txt:10]:
```suggestion
set_property(TARGET ${CMAKE_PROJECT_NAME} PROPERTY CXX_STANDARD 20)
```
Keep it aligned with the package need.

**Review by uilianries — COMMENTED** [review]
Almost good to go

File: recipes/beauty/all/conanfile.py
Comment (dfleury2) [reply recipes/beauty/all/conanfile.py:3]: Done

**Review by uilianries — APPROVED** [review]

File: recipes/beauty/all/conanfile.py
Comment (SSE4) [reply recipes/beauty/all/conanfile.py:76]: for`conan.tools.cmake` it's not needed, so it's fine to keep as is.

**Review by SSE4 — APPROVED** [review]

File: recipes/beauty/all/test_package/test_package.cpp
Comment (SSE4) [inline recipes/beauty/all/test_package/test_package.cpp:6]: in general, it's preferred to actually call some functions, because unused objects are subject for the removal by compiler (even if they have side-effects inside ctor). but it's fine to leave for now, no need to edit.

**Review by SSE4 — APPROVED** [review]

File: recipes/beauty/all/test_package/test_package.cpp
Comment (dfleury2) [reply recipes/beauty/all/test_package/test_package.cpp:6]: Ok. I note it

**Pull request merged**
